cmd/fleet/cron.go

```diff
@@ -1494,3 +1495,26 @@ func newMaintainedAppSchedule(
 
 	return s, nil
 }
+
+func newRefreshVPPAppVersionsSchedule(
+	ctx context.Context,
+	instanceID string,
+	ds fleet.Datastore,
+	logger kitlog.Logger,
+) (*schedule.Schedule, error) {
+	const (
+		name            = string(fleet.CronRefreshVPPAppVersions)
+		defaultInterval = 24 * time.Hour
+	)
+
+	logger = kitlog.With(logger, "cron", name)
+	s := schedule.New(
+		ctx, name, instanceID, defaultInterval, ds, ds,
+		schedule.WithLogger(logger),
+		schedule.WithJob("refresh_vpp_app_version", func(ctx context.Context) error {
+			return vpp.RefreshVersions(ctx, ds)
+		}),
+	)
```

Comment:
Do we have to worry about anything in the failure case here? Send an alert somewhere? Retry? Or is it acceptable for this data to be out of date for 24 hrs?

Reply:
IMO it's ok for it to be a day out of date, but will defer to @noahtalerman or @mostlikelee on that one.
I don't think that additional alerting is needed since this is pretty low stakes, but happy to add that as well if needed!

Reply:
@jahzielv is this about Fleet showing the right version on the Software title page?

![Screenshot 2025-02-14 at 9 52 02 AM](https://github.com/user-attachments/assets/f6cb1b51-d694-4439-955c-f59c79b46361)

It can take up to 24 hours for Fleet to show the right version?

Reply:
@noahtalerman yep! The spec in the bug ticket was to pull the new versions once per day; Konstantin's point is that if this job fails for whatever reason, we'd wait until the next execution (24hr) before trying again.

I think this is OK, since that's the pattern we have for other cron jobs (if they fail, we just try again next time they're scheduled), but wanted to get your input.

Reply:
@noahtalerman good idea! I think we can do that. I can chew on how to implement it.

Reply:
@noahtalerman yeah I think this can work. It doesn't look like we're using the `latest_version` field for anything other than UI display, so we could get rid of it from the database and use a call to iTunes instead to fill that field in our API handlers. 

I'll raise this to the team at standup in a bit to double check that it's a good approach.